lib/http.js

```diff
@@ -86,6 +86,8 @@ export async function addRoutes({app, service} = {}) {
 
         const {checks} = options;
         _validateChecks({checks});
+        const ecdsaKeyTypes = getEcdsaKeyTypes({credential});
+        const suite = createSuites({ecdsaKeyTypes});
```

Comment:
This isn't correct -- we should never create the suites based on the proof types, as those are provided by a potential attacker. The supported suites should instead be based on the verifier's configuration itself.

Reply:
@dlongley I'm not sure how to proceed, I've removed the use of `getEcdsaKeyTypes()`. We need a way to specify the algorithm that the ecdsa suite should use when creating the suite.  https://github.com/digitalbazaar/bedrock-vc-verifier/blob/9b8927dc02224191cfb74008c219a9dacd66bdb4/lib/suites.js#L40-L43,  otherwise the suite defaults to using the algorithm P-256, resulting in the following error:
```
"The verifier's algorithm \"P-384\" does not match the required algorithm for the cryptosuite \"P-256\"."
``` 

Here's what the verifier configuration looks like:
```
{
  "id": "https://localhost:18443/verifiers/z1AAEQYddPKxH7RbLKxFJeUmS",
  "sequence": 0,
  "controller": "did:key:z6Mko4QbcRSTBUFiPMU1PfrNM1KWK8qoSef263DfngFohXgZ",
  "meterId": "https://localhost:18443/meters/z19kgdGab9f8TbDeQszYLnZCw",
  "zcaps": {
    "edv": {
      "@context": [
        "https://w3id.org/zcap/v1",
        "https://w3id.org/security/suites/ed25519-2020/v1"
      ],
      "id": "urn:uuid:ddab8fe6-fd03-4e26-9b48-90f2b6ed8dc0",
      "controller": "did:key:z6MkibWNLTaMvjmSWafbGPbKRBtDqEWTLkK9r3AB2EMUWpTV",
      "parentCapability": "urn:zcap:root:https%3A%2F%2Flocalhost%3A18443%2Fedvs%2Fz1A7GyZGzEaoFXK3a9Tzu6Lwm",
      "invocationTarget": "https://localhost:18443/edvs/z1A7GyZGzEaoFXK3a9Tzu6Lwm",
      "expires": "2023-11-03T17:04:18Z",
      "proof": {
        "type": "Ed25519Signature2020",
        "created": "2023-11-03T16:59:18Z",
        "verificationMethod": "did:key:z6Mko4QbcRSTBUFiPMU1PfrNM1KWK8qoSef263DfngFohXgZ#z6Mko4QbcRSTBUFiPMU1PfrNM1KWK8qoSef263DfngFohXgZ",
        "proofPurpose": "capabilityDelegation",
        "capabilityChain": [
          "urn:zcap:root:https%3A%2F%2Flocalhost%3A18443%2Fedvs%2Fz1A7GyZGzEaoFXK3a9Tzu6Lwm"
        ],
        "proofValue": "z3bbahYvUxm8GRrzDEeb7fw5yeYEbueiGb3gLLj3JEVUvnuBEuBxgm5rDnCnb97yffJRatPfSiqgY3jcYU2ppDLic"
      }
    },
    "hmac": {
      "@context": [
        "https://w3id.org/zcap/v1",
        "https://w3id.org/security/suites/ed25519-2020/v1"
      ],
      "id": "urn:uuid:12b14e1f-be96-4789-be88-a138a488455e",
      "controller": "did:key:z6MkibWNLTaMvjmSWafbGPbKRBtDqEWTLkK9r3AB2EMUWpTV",
      "parentCapability": "urn:zcap:root:https%3A%2F%2Flocalhost%3A18443%2Fkms%2Fkeystores%2Fz19u97yg5uCxdsZVRQvq5pGK2",
      "invocationTarget": "https://localhost:18443/kms/keystores/z19u97yg5uCxdsZVRQvq5pGK2/keys/z19sFE18E7Yf5Nhh77xZXQbnZ",
      "expires": "2023-11-03T17:04:18Z",
      "proof": {
        "type": "Ed25519Signature2020",
        "created": "2023-11-03T16:59:18Z",
        "verificationMethod": "did:key:z6Mko4QbcRSTBUFiPMU1PfrNM1KWK8qoSef263DfngFohXgZ#z6Mko4QbcRSTBUFiPMU1PfrNM1KWK8qoSef263DfngFohXgZ",
        "proofPurpose": "capabilityDelegation",
        "capabilityChain": [
          "urn:zcap:root:https%3A%2F%2Flocalhost%3A18443%2Fkms%2Fkeystores%2Fz19u97yg5uCxdsZVRQvq5pGK2"
        ],
        "proofValue": "z53372c885CbtD5MJ3miD32LBLKpNHzD93z9VyfjPjZUQHkGxsKifigx128EFQMF4UFqvWqyNgL2BGuCxCL9W9sta"
      }
    },
    "keyAgreementKey": {
      "@context": [
        "https://w3id.org/zcap/v1",
        "https://w3id.org/security/suites/ed25519-2020/v1"
      ],
      "id": "urn:uuid:60d805a1-6e71-4a54-ad95-6a6566b18eb8",
      "controller": "did:key:z6MkibWNLTaMvjmSWafbGPbKRBtDqEWTLkK9r3AB2EMUWpTV",
      "parentCapability": "urn:zcap:root:https%3A%2F%2Flocalhost%3A18443%2Fkms%2Fkeystores%2Fz19u97yg5uCxdsZVRQvq5pGK2",
      "invocationTarget": "https://localhost:18443/kms/keystores/z19u97yg5uCxdsZVRQvq5pGK2/keys/z1A5mecTPaz2ZwBpDd5sRC1C3",
      "expires": "2023-11-03T17:04:18Z",
      "proof": {
        "type": "Ed25519Signature2020",
        "created": "2023-11-03T16:59:18Z",
        "verificationMethod": "did:key:z6Mko4QbcRSTBUFiPMU1PfrNM1KWK8qoSef263DfngFohXgZ#z6Mko4QbcRSTBUFiPMU1PfrNM1KWK8qoSef263DfngFohXgZ",
        "proofPurpose": "capabilityDelegation",
        "capabilityChain": [
          "urn:zcap:root:https%3A%2F%2Flocalhost%3A18443%2Fkms%2Fkeystores%2Fz19u97yg5uCxdsZVRQvq5pGK2"
        ],
        "proofValue": "z3NddmjmvLezLA3dGZMBGwjceMcQhPdHxNmzhfUDTSUwXiSD4v2ZRNY9J7XcYWbyRiQRnNB4LSCjYxmxHPqk636yg"
      }
    }
  }
}
```

Reply:
@JSAssassin,

The problem is with the cryptosuite. It needs to accept both P-256 and P-384. So we've got a lower layer issue to resolve.